### ISSUE TITLE
factory pattern

### ISSUE THREAD

**Issue by sol-myob** [monajena27/Toy-Block-Factory#11]
https://github.com/monajena27/Toy-Block-Factory/blob/49a364f7e6039d2b1178996a5e9b2f91c94a6959/ToyBlockFactoryKata/Tables/ITableGenerator.cs#L7

This is actually more the factory pattern, so renaming generator to factory for these would make sense

**Comment by mona-jena** [comment]
wouldn't the same logic that applied to the next issue for PricingCalculator also apply to ITableGenerator, technically it's an interface for the generators?

**Comment by sol-myob** [comment]
Agreed .  But calculate is the right word for the operation and it doesn't come overloaded. The overloadedness is my main apprehension.

Generators are a common programming technique and they are what I think of when I see generator anywhere in code

**Comment by mona-jena** [comment]
ohh I didn't know that

**Comment by mona-jena** [comment]
So would the names for the diff tables also change to ColourTableFactory and QuantityableFactory?

**Comment by sol-myob** [comment]
There are a few patterns which are considered "creational". I would figure which one fits the pattern here best, and then use that:

https://en.wikipedia.org/wiki/Creational_pattern

it may be possible to name each one in a way where you dont need to suffix with the pattern. but that's often more of a struggle to get right
